src/main/java/com/coveros/selenified/services/HTTP.java

```diff
@@ -291,11 +286,12 @@ private Response call(String call, String service, Request request, File file) {
                 connection.setRequestProperty("Authorization", "Basic " + encoding);
             }
             connection.connect();
-            if ((request != null && request.getData() != null) || file != null) {
-                if (connection.getRequestProperty(CONTENT_TYPE).startsWith("application/json;")) {
-                    writeJsonDataRequest(connection, request);
-                } else if (connection.getRequestProperty(CONTENT_TYPE).startsWith("multipart/form-data;")) {
+            if ((request != null && request.isPayload()) || file != null) {
+                if (connection.getRequestProperty(CONTENT_TYPE).startsWith("multipart/form-data") ||
+                        request.getMultipartData() != null || file != null) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "request" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS2259)

Reply:
I think you might be reading it wrong, it looks like the if statements on 268 and 269 would allow request to be null if file was not null, because of your ||.